### PR TITLE
{Core} Add util function highlight_command

### PR DESCRIPTION
**Pull request by houk-ms**
**Description**

This PR adds an util function to highlight a command.

The function may be used in
- error recommendation
- az next

**Testing Guide**

**History Notes**

[Component Name 1] BREAKING CHANGE: az command a: Make some customer-facing breaking change.
[Component Name 2] az command b: Add some customer-facing feature.

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [ ] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

File: src/azure-cli-core/azure/cli/core/style.py
Comment (jiasli) [inline src/azure-cli-core/azure/cli/core/style.py:79]: Some example can be very helpful, like

https://github.com/Azure/azure-cli/blob/903ecf1c0e75cb1fde19c6be0ca22c52b35a7ef9/src/azure-cli-core/azure/cli/core/util.py#L1186-L1190

**Comment by jiasli** [comment]
The style framework is far from perfect. I made several refinement in #16258. I suggest we put this PR on hold until #16258 is fully tested and merged.

File: src/azure-cli-core/azure/cli/core/style.py
Comment (fengzhou-msft) [inline src/azure-cli-core/azure/cli/core/style.py:97]: Can you add comment about the rule used?

Comment (jiasli) [reply src/azure-cli-core/azure/cli/core/style.py:97]: I made some comments at https://github.com/Azure/azure-cli/pull/16257#discussion_r547698672

The logic of `=` handling is a little bit tricky. It works for `az config set a=b`, but it doesn't work for `az config unset output`. It  is the nature of positional argument that we can NOT decide whether an arg is 
- part of a command, or
- a positional argument 

until semantic analysis is involved. 

Let's keep it simple and treat it as part of the command by now.

Comment (jiasli) [reply src/azure-cli-core/azure/cli/core/style.py:97]: Another possible solution is to utilize `linter_exclusions.yml` which contains the full list of commands with positional arguments:

https://github.com/Azure/azure-cli/blob/2c52cbcef7f43c85088118e672ded491d674737d/linter_exclusions.yml#L1181-L1185

File: src/azure-cli-core/azure/cli/core/style.py
Comment (houk-ms) [reply src/azure-cli-core/azure/cli/core/style.py:79]: Added more comments about the highlighting rules. The return type is of list of styled text, not easy to provide example.

File: src/azure-cli-core/azure/cli/core/style.py
Comment (houk-ms) [reply src/azure-cli-core/azure/cli/core/style.py:104]: This might be a solution, but also making the function much more complex and harder to maintain. 

I prefer we keep it simple, until we really need to consider the case.

File: src/azure-cli-core/azure/cli/core/style.py
Comment (jiasli) [inline src/azure-cli-core/azure/cli/core/style.py:180]:
```suggestion
    :rtype: list
```

**Review by zhoxing-ms — APPROVED** [review]

File: src/azure-cli-core/azure/cli/core/style.py
Comment (jiasli) [inline src/azure-cli-core/azure/cli/core/style.py:175]:
```suggestion
    """Highlight a command with colors.

    For example, for 
        
        az group create --name myrg --location westus

    The command name 'az group create', argument name '--name', '--location' are marked as ACTION style.
    The argument value 'myrg' and 'westus' are marked as PRIMARY style.
    If the argument is provided as '--location westus', it will be marked as PRIMARY style.
```

File: src/azure-cli-core/azure/cli/core/style.py
Comment (jiasli) [inline src/azure-cli-core/azure/cli/core/style.py:180]:
```suggestion
    :param raw_command: The command that needs to be highlighted.
    :type raw_command: str
    :return: The styled command text.
    :rtype: list
```
    :rtype: list
```

**Review by fengzhou-msft — APPROVED** [review]

**Review by jiasli — APPROVED** [review]

**Pull request merged**
